Forge.TreeWalker/src/ForgeSchemaValidator.cs

```diff
@@ -0,0 +1,263 @@
+        }
+
+        /// <summary>
+        /// Validates single or multiple schemas from input path with the given rules.        
+        /// </summary>
+        /// <param name="path">The path that contains a schema file</param>
+        /// <param name="rules">The rules used to validate input schemas is only allowed in string or JSchema type</param>
+        /// <param name="validateAsDictionary">True if the custom rules is to handle the whole dictionary</param>
+        /// <returns>The result of schema validation. The errorList would contain error message if validation fails</returns>
+        public static bool ValidateSchemaFromPath(string path, object rules, bool validateAsDictionary, out IList<ValidationError> errorList)
+        {
+            List<JObject> schemas = GetSchemaFromPath(path, validateAsDictionary);
+
+            return Validate(schemas, rules, out errorList);
+        }
+
+        /// <summary>
+        /// Validates single or multiple schemas from input path with the given rules.
+        /// </summary>
+        /// <param name="path">The path that contains a schemas directory</param>
+        /// <param name="rules">The rules used to validate input schemas is only allowed in string or JSchema type</param>
+        /// <param name="validateAsDictionary">True if the custom rules is to handle the whole dictionary</param>
+        /// <returns>The result of schema validation. The errorList would contain error message if validation fails</returns>
+        public static bool ValidateSchemaFromDirectory(string path, object rules, bool validateAsDictionary, out IList<ValidationError> errorList)
+        {
+            List<JObject> schemas = GetAllSchemasInDirectory(path, validateAsDictionary);
+
+            return Validate(schemas, rules, out errorList);
+        }
+
+
+        private static List<JObject> ConvertDictionaryToJObjectList(Dictionary<string, ForgeTree> schemas, bool validateAsDictionary)
+        {
+            List<JObject> schemaList = new List<JObject>();
+
+            if (validateAsDictionary)
+            {
+                schemaList.Add(SerializeToJObject(schemas));
+            }
+            else
+            {
+                foreach (ForgeTree item in schemas.Values)
+                {
+                    schemaList.Add(SerializeToJObject(item));
+                }
+            }
+
+            return schemaList;
+        }
+
+        private static List<JObject> ConvertStringToJObjectList(string schema, bool validateAsDictionary)
+        {
+            List<JObject> schemaList = new List<JObject>();
+
+            //There could be three possible cases:
+            //1. TreeName mapped to the ForgeTree in the dictionary and custom rules handle dictionary.
+            //2. There are only ForgeTree without matching forge tree name custom rules handle ForgeTree list.
+            //3. The string schema that could be deserialized as a ForgeTree should be parsed directly
```

Comment:
// 1. schema is a Dictionary<string, ForgeTree> and should be validated as a dictionary.
// 2. schema is a Dictionary<string, ForgeTree> and should be validated as individual ForgeTree's.
// 3. schema is a ForgeTree and shoudl be validated as a single ForgeTree.